client/my-sites/upgrades/domain-management/dns/dns-list.jsx

```diff
@@ -43,7 +43,11 @@ const DnsList = React.createClass( {
 	addDns: function( record ) {
 		addDnsAction( this.props.selectedDomainName, record, ( error ) => {
 			if ( error ) {
-				notices.error( error.message );
+				notices.error( error.message || this.translate( 'The DNS record could not be restored.' ) );
+			} else {
+				this.props.successNotice( this.translate( 'The DNS record has been restored.' ), {
+					duration: 5000
```

Comment:
We are already using inline form validation errors in our forms and migrating the older forms. 

I think you're trying the solve the problem in the wrong level; making something hard for the developers won't result in a better user experience.

Also, sometimes there isn't enough space put a notice near the component without breaking the whole layout; a global, overlaying notice solve the problem right away. If the global notice component is not getting enough attention, again it's a design problem and our designers can take look at it. 

re: Render tree: Yes, technically there will be same amount of elements in the DOM tree, but by using it within the component, we're duplicating the code.